salt/utils/url.py

```diff
@@ -47,7 +47,7 @@ def create(path, saltenv=None):
 
     query = f"saltenv={saltenv}" if saltenv else ""
     url = salt.utils.data.decode(urlunparse(("file", "", path, "", query, "")))
```

Comment:
I recommend just not using schema when calling `unparse`/`unsplit`:

```suggestion
   return f"salt://{salt.utils.data.decode(urlunsplit(("", "", path, query, "")))}"
```

Reply:
I just read through the source code of `urlunparse`/`urlunsplit` and it looks like there is not currently any special handling for the file scheme so I agree that this is probably a better solution.

Reply:
Ok, lets see how this works. Tests on Python 3.12.6 are green with this version, I'll let CI check older ones.

Reply:
Using urlunparse is okay, but I think using `urlunsplit` as @yangskyboxlabs is more ideal since it passes one less unused argument.